### PR TITLE
[ldap-users] remove threading

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-3547

instead of searching user by users, we get all users and find missing ones in memory.

### CODE REVIEW

File: reconcile/utils/ldap_client.py [hunk 1]
```diff
@@ -27,11 +27,15 @@ def init_from_config():
...
 
     with init_from_config() as client:
-        result, _, _, _ = client.search(_base_dn,
-                                        f'(&(uid={username})' +
-                                        '(objectclass=person))')
-        return result
+        _, _, results, _ = client.search(_base_dn, '(&(objectclass=person))')
+        users = set()
+        for r in results:
+            dn = r['dn']
```
Comment (Piojo) [inline reconcile/utils/ldap_client.py:37]: Just extract the uid field here.
```suggestion
            uid = r['uid']
```

**Review by Piojo — CHANGES_REQUESTED** [review]
Please note that, while all my comments are correct from an LDAP perspective, I am not familiar with the Python modules - the return structure of `search` may need some adjustments. API docs: https://ldap3.readthedocs.io/en/latest/searches.html

**Comment by Piojo** [comment]
I'd be curious to know the new timings. :)

File: reconcile/utils/ldap_client.py [hunk 2]
```diff
@@ -27,11 +27,15 @@ def init_from_config():
...
-def user_exists(username):
+def get_users():
     global _base_dn
 
     with init_from_config() as client:
-        result, _, _, _ = client.search(_base_dn,
-                                        f'(&(uid={username})' +
-                                        '(objectclass=person))')
-        return result
+        _, _, results, _ = client.search(_base_dn, '(&(objectclass=person))')
```
[the remark this answers is not in the document]
Comment (maorfr) [reply reconcile/utils/ldap_client.py]: added `attributes=["uid"]`, thanks! there is no significant difference passing uids or not (4-5 seconds for both ways).

File: reconcile/utils/ldap_client.py (hunk 1 above)
Comment (maorfr) [reply reconcile/utils/ldap_client.py]: thanks!

**Comment by maorfr** [comment]
time improvement: 2m39s -> 8s (!!!)

**Review by Piojo — APPROVED** [review]
LGTM. A future optimization is to query only for the 700-or-so users known to app-interface instead of the ~10K IDs we may get from querying the entire LDAP subtree. But for now, this is awesome and I bet the LDAP owners will be pleased as well.

**Comment by maorfr** [comment]
changed the query to only be for the known users, as you suggested. run time is still ~8s, but it's obviously better this way.

**Pull request merged**